Description: Drop targets option to force .browserslistrc file
Author: Pirate Praveen <praveen@debian.org>
Bug-Debian: https://bugs.debian.org/991656
Forwarded: not-needed
Reviewed-By: Yadd <yadd@debian.org>
Last-update: 2021-07-29

Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=991656

--- a/.lib.babelrc
+++ b/.lib.babelrc
@@ -1,11 +1,3 @@
 {
-  "presets": [
-    ["@babel/env", {
-      "targets": {
-        "node": "6.4.0",
-        "browsers": [ "> 1%", "last 2 versions", "ie 9" ]
-      }
-    }]
-  ],
   "sourceMaps": "inline"
 }
